RSCBC-164, RSCBC-165: Rustify the docs

chvck opened the pull request.

chvck committed:
RSCBC-164, RSCBC-165: Rustify the docs

RichardSmedley commented on modules/devguide/examples/striptags.sh:
This is used to copy across to the GH devguide repo (now known as sdk-examples),
which Ray is looking to revive.

I'll put it back in a later PR.

RichardSmedley commented on modules/howtos/pages/kv-operations.adoc:
```diff
@@ -16,327 +16,193 @@ See xref:concept-docs:documents.adoc[the concept guide to _Documents_] for a dee
-reactive API.  See xref:concurrent-async-apis#choosing-an-api[Choosing an API] for more details.
-=====
-
 == Handling Single Errors
```
No point in keeping this header for one admonition - in fact, let's bring the header into the admonition...

RichardSmedley committed:
oving heading into admonition

RichardSmedley committed:
Em dash

RichardSmedley committed:
Gardening

RichardSmedley commented on modules/howtos/pages/provisioning-cluster-resources.adoc:
```diff
@@ -4,67 +4,39 @@
 include::{version-common}@sdk:shared:partial$flush-info-pars.adoc[tag=management-intro]
 
-The Scala SDK also comes with some convenience functionality for common Couchbase management requests.
+The {name-sdk} also comes with some convenience functionality for common Couchbase management requests.
```
Not sure if the management API is unimplemented, and this page should be removed,
or if this is a future todo -
so I will put page through for now.

RichardSmedley commented on modules/howtos/pages/sqlpp-queries-with-sdk.adoc:
```diff
@@ -2,25 +2,16 @@
-:lang: Scala
-:example-source: devguide:example$scala/Queries.scala
-:example-source-lang: scala
+:lang: Rust
```
Do we need this?
Not sure why it got introduced, but I'm not convinced that it's useful.

RichardSmedley commented on modules/howtos/pages/sqlpp-queries-with-sdk.adoc:
```diff
@@ -32,135 +23,64 @@ include::howtos:partial$n1ql-getting-started.adoc[]
 Note that building indexes is covered in more detail on the xref:concept-docs:n1ql-query.adoc#index-building[Query concept page] --
 and in the https://docs.couchbase.com/sdk-api/couchbase-scala-client/com/couchbase/client/scala/manager/query/index.html[API Reference].
```
is there a pre-release API ref I can get links from?

chvck replied:
Not until I publish the developer preview

RichardSmedley committed:
Greenfield site - no aliases necessary

RichardSmedley committed:
Obscure grammar rule

(One that's usually done this way - correctly - in US English :us:  :raised_eyebrow: )

RichardSmedley commented on modules/howtos/pages/vector-searching-with-sdk.adoc:
```diff
@@ -14,9 +14,6 @@
-
-
 ////
 * Couchbase Server 7.6 introduces Vector Search to enable AI integration, semantic search, and the RAG framework.
```
I'll try and add Vector GSI examples in time for Morpheus.

RichardSmedley commented on modules/howtos/partials/n1ql-intro.adoc:
```diff
@@ -1,6 +1,6 @@
 // Required attributes:
```
I think these partials need a revisit from me,
to see if this abstracion still has a needed role

RichardSmedley commented on modules/project-docs/pages/compatibility.adoc:
```diff
@@ -169,80 +98,48 @@ include::{version-common}@sdk:shared:partial$capella.adoc[tag=cloud]
+[cols="7,6"]
 |===
-| | Server 7.0 & 7.1 | Server 7.2 | Server 7.6
+| | Server 7.6
```
I'm going to put this through as-is,
but I'll be revisiting it before we publish...

chvck replied:
I believe PM are ok with rust supporting 7.6+ but should probably check that...

RichardSmedley reviewed (approved):
Thanks @chvck - sterling work :+1:
